C_CPP/Algorithmic Approaches/Sorting/sorting.md

```diff
@@ -0,0 +1,252 @@
+## what is sorting?
+
+sorting basically means to arrange the data items in increasing or decreasing manner according to the relationship  among the data's.
+> FOR EXAMPLE - we have given an array [9,5,7,3,1]
+> Now let us suppose we need to sort the array in increasing/ascending order , so it will become 
+> array after sorting - [1,3,5,7,9] (ascending).This will give us data in a sorted manner.
+
+## why do we need sorting and it's different methods?
+
+ - If we have a huge data with us (suppose 1 lakh entries) and we are provided with a task to select all the person's in that data whose roll no starts with 1 upto 100 . Now in this if we don't have any sorting technique then we need to go through all the data (i.e approx lakh) to select the needful,and we need to go through it several times to arrange all data in order.
+Therefore the solution of above complication is **_SORTING TECHNIQUE_**.
+ - Time complexity is the main advantage of sorting.problems can be solved easily or quickly based on sorting.
+
+
+## types of sorting 
+
+> - **Bubble sort** - It is a simple and basic algorithm used for sorting.This is a comparison-based algorithm in which each pair of adjacent elements is compared and the elements are swapped if they are not in order.
```

Comment:
Instead of defining here, add a brief description just above your code.

Also, add relevant comments in your code to make it easily understandable.